src/dmd/e2ir.d

```diff
@@ -3123,6 +3123,36 @@ elem *toElem(Expression e, IRState *irs)
                         el_free(e1);
                         return setResult2(e);
                     }
+
+                    static bool allZeroBits(ref Expressions exps)
+                    {
+                        foreach (e; exps[])
+                        {
+                            /* The expression types checked can be expanded to include
+                             * floating point, struct literals, and array literals.
+                             * Just be careful to return false for -0.0
+                             */
+                            if (!e ||
+                                e.op == TOK.int64 && e.isIntegerExp().toInteger() == 0 ||
+                                e.op == TOK.null_)
+                                continue;
+                            return false;
+                        }
+                        return true;
+                    }
+
+                    /* Use a memset to 0
+                     */
+                    if ((sle.useStaticInit ||
+                         sle.elements && allZeroBits(*sle.elements) && !sle.sd.isNested()) &&
+                        ae.e2.type.isZeroInit(ae.e2.loc))
```

Comment:
Should this be
```d
(sle.useStaticInit && ae.e2.type.isZeroInit(ae.e2.loc))
||
(sle.elements && allZeroBits(*sle.elements) && !sle.sd.isNested())
```
?

Reply:
no because sle.elements does not include default initializers.